### PR TITLE
Reorganize piccr return output

### PR DESCRIPTION
**Pull request by thomas-muench**
This PR suggests all changes needed in order to set up the return output of piccr to work with Cpt-Picarr; see #17.

### CODE REVIEW

File: tests/testthat/testExampleConfig.R [hunk 1]
```diff
@@ -27,16 +27,17 @@ test_that("test return value and outputs", {
...
   
   # check format of return value
-  expect_length(processedData, 4)
-  expect_length(processedData$memoryCorrected, 3)
-  expect_length(processedData$calibrated, 3)
-  expect_length(processedData$processed, 3)
-  expect_length(processedData$pooledStdDev, 3)
+  expect_true(is.list(processedData))
+  expect_length(processedData, 3)
+  #expect_length(processedData$memoryCorrected, 3)
```
Comment (twollnik) [inline tests/testthat/testExampleConfig.R]: let's add a TODO comment. `# TODO: uncomment these tests when the new interface is stable`. That way we can find this section when searching for "TODO".

File: R/main_processData.R [hunk 2]
```diff
@@ -19,24 +19,51 @@
...
+    pooledSD = NULL,
+    memoryCoefficients = NULL,
+    calibrationParams = NULL,
+    driftParams = NULL
+  )
+
+  output <- list()
+  for (i in 1 : length(datasets)) {
+    output[[i]] <- outputTemplate
+  }
```
Comment (twollnik) [inline R/main_processData.R:65]: We can use `rep` or `map` to simplify this section.

```R
# using rep:
output <- rep(list(outputTemplate), length(datasets))

# using map
output <- map(datasets, ~ outputTemplate)
```

File: R/main_processData.R [hunk 3]
```diff
@@ -62,9 +89,32 @@ processData <- function(datasets, config){
...
+
+  for (i in 1 : length(datasets)) {
+
+    output[[i]]$name <- namesOfDatasets[i]
+    
+    output[[i]]$raw <- datasets[[i]]
+    
+    output[[i]]$processed <- processedData[[i]]
+    
+  }
```
Comment (twollnik) [inline R/main_processData.R:122]: This for loop could be replaced with the following code. If we do it this way we don't even need to initialize the output. (We could get rid of lines 41 to 65.)

```R
output <- map(names(datasets), function(name){
  list(
    name = name,
    raw = datasets[[name]],
    processed = processedData[[name]]
  )
})
```

**Review by twollnik — COMMENTED** [review]
Good work overall, thanks for opening this PR. I added some suggestions on how the code may be written in a more concise way.

File: R/main_processData.R (hunk 3 above)
Comment (thomas-muench) [reply R/main_processData.R:122]: Thanks for the suggestion! If we do it this way, we need if-else constructions for the switchable processes, - such as memory correction: if (correction) fill with values else fill with NULL - to ensure the output structure to always be complete, right?

Comment (twollnik) [reply R/main_processData.R:122]: That would be one good way to do it. Alternatively we can always set the default in the map and use an if construct to modify the output afterwards.

File: tests/testthat/testExampleConfig.R (hunk 1 above)
Comment (thomas-muench) [reply tests/testthat/testExampleConfig.R]: Good idea; I commented out the failing tests and added TODO flags: 47f226b728475513ad5c5977cc1b9c6720874f1c.

**Comment by twollnik** [comment]
@thomas-muench Thanks for all your work on this task. Merging this PR now.

**Pull request merged**